presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/PhoenixClient.java

```diff
@@ -0,0 +1,453 @@
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.timeWriteFunction;
+import static io.prestosql.plugin.jdbc.StandardColumnMappings.varcharColumnMapping;
+import static io.prestosql.plugin.phoenix.MetadataUtil.toPhoenixSchemaName;
+import static io.prestosql.plugin.phoenix.PhoenixErrorCode.PHOENIX_METADATA_ERROR;
+import static io.prestosql.plugin.phoenix.PhoenixErrorCode.PHOENIX_QUERY_ERROR;
+import static io.prestosql.plugin.phoenix.PhoenixMetadata.DEFAULT_SCHEMA;
+import static io.prestosql.plugin.phoenix.TypeUtils.getArrayElementPhoenixTypeName;
+import static io.prestosql.plugin.phoenix.TypeUtils.getJdbcObjectArray;
+import static io.prestosql.plugin.phoenix.TypeUtils.jdbcObjectArrayToBlock;
+import static io.prestosql.plugin.phoenix.TypeUtils.toBoxedArray;
+import static io.prestosql.spi.StandardErrorCode.NOT_SUPPORTED;
+import static io.prestosql.spi.type.DoubleType.DOUBLE;
+import static io.prestosql.spi.type.RealType.REAL;
+import static io.prestosql.spi.type.TimeType.TIME;
+import static io.prestosql.spi.type.TimeWithTimeZoneType.TIME_WITH_TIME_ZONE;
+import static io.prestosql.spi.type.TimestampWithTimeZoneType.TIMESTAMP_WITH_TIME_ZONE;
+import static io.prestosql.spi.type.VarcharType.createUnboundedVarcharType;
+import static java.lang.String.format;
+import static java.lang.String.join;
+import static java.sql.Types.ARRAY;
+import static java.sql.Types.FLOAT;
+import static java.sql.Types.LONGNVARCHAR;
+import static java.sql.Types.LONGVARCHAR;
+import static java.sql.Types.NVARCHAR;
+import static java.sql.Types.TIMESTAMP;
+import static java.sql.Types.TIMESTAMP_WITH_TIMEZONE;
+import static java.sql.Types.TIME_WITH_TIMEZONE;
+import static java.sql.Types.VARCHAR;
+import static java.util.Collections.nCopies;
+import static java.util.Objects.requireNonNull;
+import static java.util.function.Function.identity;
+import static org.apache.phoenix.coprocessor.BaseScannerRegionObserver.SKIP_REGION_BOUNDARY_CHECK;
+import static org.apache.phoenix.query.QueryConstants.NULL_SCHEMA_NAME;
+import static org.apache.phoenix.util.SchemaUtil.ESCAPE_CHARACTER;
+
+public class PhoenixClient
+        extends BaseJdbcClient
+{
+    private final Configuration configuration;
+
+    @Inject
+    public PhoenixClient(PhoenixConfig config)
+            throws SQLException
+    {
+        super(
+                new BaseJdbcConfig(),
+                ESCAPE_CHARACTER,
+                new DriverConnectionFactory(
+                        DriverManager.getDriver(config.getConnectionUrl()),
+                        config.getConnectionUrl(),
+                        Optional.empty(),
+                        Optional.empty(),
+                        config.getConnectionProperties()));
+        requireNonNull(config, "config is null");
+        configuration = new Configuration(false);
+        ConnectionInfo connectionInfo = PhoenixEmbeddedDriver.ConnectionInfo.create(config.getConnectionUrl());
```

Comment:
Nit: the `PhoenixEmbeddedDriver.` prefix is redundant since `ConnectionInfo` is already imported directly